### PR TITLE
Mocks rollout and implementation of testing behaviors

### PR DESCRIPTION
**Pull request by AlexVCaron**
- Make apply-mocks fixture usable by all tests.
- Start hooks and fixtures modules for pytest, loaded using conftest.py in the script tests directory.

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/1/

**Comment by frheault** [comment]
Is it possible to have a comparison of timing before and after / with and without MOC?

The code looks good, @arnaudbore do you think we can merge my two PR (python3.10 and the fix) next week and then Alex could go forward with this PR. And as soon as it is working we update the test data, and do a release candidate?

I tested the 4 main flow once, I did not try Freewater. I think I will re-test with a RC1.

**Comment by frheault** [comment]
Also, @AlexVCaron can you describe what is in the new archive? What is different?

**Comment by AlexVCaron** [comment]
> Also, @AlexVCaron can you describe what is in the new archive? What is different?

TLDR : files are practically the same, but the new ones are the right datatype so comparison of outputs to expectations works for both mocked and unmocked cases.

Three outputs files awaited by the tests have slightly changed to align with unmocked tests, namely :
- fodf_descoteaux07_sub_full.nii.gz
- fodf_descoteaux07_sub_sym.nii.gz
- fodf_descoteaux07_sub_twice.nii.gz

It worked when mocking, since we only check if processing errors were done by the script on those files, but didn't when running the full algorithm on them.

The difference obtained between the output and the one awaited was tiny and after investigations, I think it was because of a type conversion error on my end.

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/2/

**Comment by AlexVCaron** [comment]
I started thinking about mocks again. Where do we go from here ? There is a few scripts that would benefit from mocking, but we need to address how and when we use the through Jenkins. I also created a sheet to follow improvements that come from using mocking [here](https://docs.google.com/spreadsheets/d/1xOR24wYa5wmURXoqiQfomWIfQOGsGyUnsqZqG2ng4ck/edit?usp=sharing). I'll update it when new mocks come into action.

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/3/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/4/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/5/

**Comment by EmmaRenauld** [comment]
We should talk about this at our retreat preparation!

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/6/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/8/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/10/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/11/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/12/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/13/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/14/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/15/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/21/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/25/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/28/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-664/29/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !
